mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +230,139 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  @transient private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
```

Comment:
If we make the whole thing `transient`, does it mean even the small object like `lossHistory` can not be saved as part of model? I'm still leaning toward having lossHistory as part of LinearRegressionModel, and link to training dataframe using @transient. In fact, in R, those small stats are part of models.